source/JsMaterialX/JsMaterialXCore/JsElement.cpp

```diff
@@ -35,11 +35,34 @@ namespace mx = MaterialX;
 
 EMSCRIPTEN_BINDINGS(element)
 {
+    ems::class_<mx::ElementEquivalenceOptions>("ElementEquivalenceOptions")
+        .constructor<>()
+        .property("performValueComparisons", &mx::ElementEquivalenceOptions::performValueComparisons)
+        .property("floatFormat", &mx::ElementEquivalenceOptions::floatFormat)
+        .property("floatPrecision", &mx::ElementEquivalenceOptions::floatPrecision)
+        .function("setAttributeExclusionList", ems::optional_override([](mx::ElementEquivalenceOptions& self, const std::vector<std::string>& exclusionList) 
+        {
+            self.attributeExclusionList = std::set<std::string>(exclusionList.begin(), exclusionList.end());
+        }));
+
     ems::class_<mx::Element>("Element")
         .smart_ptr<std::shared_ptr<mx::Element>>("Element")
         .smart_ptr<std::shared_ptr<const mx::Element>>("Element") // mx::ConstElementPtr
         .function("equals", ems::optional_override([](mx::Element& self, const mx::Element& rhs) { return self == rhs; }))
         .function("notEquals", ems::optional_override([](mx::Element& self, const mx::Element& rhs) { return self != rhs; }))
+        .function("isEquivalent", ems::optional_override([](mx::Element &self, const mx::Element& rhs, const mx::ElementEquivalenceOptions& options, 
+                                                            ems::val message) 
+        {
+            mx::ConstElementPtr rhsPtr = rhs.getSelf();
+            std::string differences;
```

Comment:
Now that our equivalence tests return a `message` string, can we potentially remove the duplication between `message` and `differences` strings in this logic?

Reply:
I've fixed this up so it is have a proper variadic binding return message argument. The ems message is an object argument and the return type from C++ is a string so is mapped to the object's `.message` member (the same as is done for `validate()`).